pyproject.toml

```diff
@@ -83,17 +83,65 @@ addopts = [
 ]
 xfail_strict = true
 
-[tool.black]
+[tool.ruff]
+# Exclude a variety of commonly ignored directories.
+exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".git-rewrite",
+    ".hg",
+    ".ipynb",
+    ".ipynb_checkpoints",
+    ".mypy_cache",
+    ".nox",
+    ".pants.d",
+    ".pyenv",
+    ".pytest_cache",
+    ".pytype",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    ".vscode",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "site-packages",
+    "venv",
+]
+
+# Same as Black.
 line-length = 120
-target-version = ['py311']
+indent-width = 4
+show-fixes = true
+
+# Assume Python 3.12
+target-version = "py312"
+
+[tool.ruff.lint]
+select = [
+    # pycodestyle
+    "E",
+    # Pyflakes
+    "F",
+    # isort
+    "I",
+]
 
-[tool.isort]
-profile = "black"
-combine_as_imports = true
-line_length = 120
+[tool.ruff.lint.isort]
+# Imports that are imported using keyword "as" and are from the same source - are combined.
+combine-as-imports = true
 
-[tool.pylint]
-max-line-length = 120
+[tool.ruff.lint.per-file-ignores]
+# Ignore `F811` (redefinition violations) in all examples notebooks since we use redefinition.
+"docs/examples/*.ipynb" = ["F811"]
+# Ignore 'E402' (import not at the top) in all docs files.
+"docs/*" = ["E402"]
```

Comment:
which other, non-example file in `docs/` has this offending thing? we may want to fix it (not in the notebooks though)

Reply:
please disable in that line  using `# noqa: E402` because it's very specific to that file